updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,90 @@
+[id="update-using-custom-machine-config-pools"]
+= Performing a rolling update 
+include::modules/common-attributes.adoc[]
+:context: update-using-custom-machine-config-pools
+
+toc::[]
+
+
+Using a rolling update strategy allows you to update groups of nodes in a cluster one-by-one, test the nodes and the applications on those nodes, and if everything works as you expect, you can proceed to update another group of nodes. 
+
+The {product-title} update process is highly-available by design and there are many features, such as pod disruption budgets, pod affinity, health checks, and replicas, which make the applications highly-available in case of node failures.
+
+However there might be some secenarios where you want a more controlled rollout of an upgrade to the worker nodes in order to ensure that mission-critical applications stay available during the whole update, even if the upgrade process causes a failure. After the cluster is updated, you can update those nodes at more appropriate time.
+
+Updating the nodes in subsets in this manner is referred to as a _rolling_ update of a _canary_ update.
+
+Using rolling update can also help with large clusters with hundreds of nodes that cannot be fully updated in your organization's defined maintenance window, or if your organization uses multiple maintenance windows. 
```

Comment:
`if your organization uses multiple maintenance windows.` does not sound correct to me. Organizations do not like to use multiple maintenance windows in general . They will be happy with no maintenance windows. But because there is a chance something might break in the whole stack during update process and they need to do preflight + post flight checks to make sure everything works as expected we need maintenance windows . The challenge is around duration of the maintenance window they can afford. We already know updating large clusters can take more than 12/18/24 hours. But it is practically impossible to take a maintenance window of 24 hours . The risk is too high.  Because if something breaks then again it would need few hours or more to restore. So our suggestion is that customers can use custom machine config pools and can take smaller maintenance windows e.g 4 hours instead of a single large maintenance window.